src/pyodide/internal/builtin_wrappers.js

```diff
@@ -18,6 +18,23 @@ export function monotonicDateNow() {
   return now + lastDelta;
 }
 
+/**
+ * We initialize Python at top level, but it tries to initialize the random seed with
+ * crypto.getRandomValues which will fail at top level. So we don't produce any entropy the first
+ * time around and we reseed the rng in the first request context before executing user code.
+ */
```

Comment:
This won't happen if we are restoring from a memory snapshot since it occurs in the call to `main()` and we skip calling `main()` in that case.